src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,115 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+const DEFAULT_FORM_VALUES = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [formValues, setFormValues] = useState<Movie>({
+    ...DEFAULT_FORM_VALUES,
+  });
+
+  const handleFormValueChange = (newValue: string, formFieldTitle: string) => {
+    setFormValues(prevState => {
+      return {
+        ...prevState,
+        [formFieldTitle]: newValue,
+      };
+    });
+  };
+
+  const isErrorsInForm =
+    formValues.title &&
+    formValues.imdbId &&
+    formValues.imgUrl &&
+    formValues.imdbUrl;
+
+  const handleClearForm = () => {
+    setFormValues({ ...DEFAULT_FORM_VALUES });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!isErrorsInForm) {
+      return;
+    }
+
+    onAdd({ ...formValues });
+    setCount(prevState => prevState + 1);
+    handleClearForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={formValues.title}
+        onChange={(newValue: string, formTitle: string) =>
+          handleFormValueChange(newValue.trimStart(), formTitle)
```

Comment:
I think, we can do it in middle of the function
```suggestion
          handleFormValueChange(newValue, formTitle)
```
```
if (!newValue.trimStart()) {
  return;
}
```
for example or something else) then we won't have extra sets